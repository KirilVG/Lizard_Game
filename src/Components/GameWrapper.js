import React, { useEffect } from "react";
import Canvas from "./Canvas";
import GameLogic from "./GameLogic";
import Canvas2DRenderer from "./Canvas2DRenderer";

const GameWrapper = (props) => {
  const { id } = props;

  useEffect(() => {
    const gameCanvas = document.getElementById(id);

    const canvas2dRenderer = new Canvas2DRenderer(gameCanvas);

    const gameLogic = new GameLogic(
      gameCanvas.width,
      gameCanvas.height,
      props.lanesNum,
      canvas2dRenderer,
      props.gameEndHandler
    );
  }, []);

  return (
    <div>
      <Canvas canvasID={id}></Canvas>
    </div>
  );
};

export default GameWrapper;
