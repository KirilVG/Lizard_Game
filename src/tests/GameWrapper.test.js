import renderer from "react-test-renderer";
import GameWrapper from "../Components/GameWrapper";

const mockedId = "Id";
const mockedLanesNum = 3;
const mockdedGameEndHandler = () => {};

it("game wrapper should be rendered correctly", () => {
  const handleGameEnd = () => {};
  const tree = renderer
    .create(
      <GameWrapper
        id={mockedId}
        lanesNum={mockedLanesNum}
        gameEndHandler={mockdedGameEndHandler}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
