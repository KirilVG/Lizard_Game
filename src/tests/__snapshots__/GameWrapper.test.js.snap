// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game wrapper should be rendered correctly 1`] = `
<div>
  <canvas
    height={760.3199999999999}
    id="Id"
    width={1024}
  />
</div>
`;
